rclpy/src/rclpy/timer.cpp

```diff
@@ -41,10 +41,12 @@ Timer::destroy()
 }
 
 Timer::Timer(
-  py::capsule pyclock, py::capsule pycontext, int64_t period_nsec)
-: clock_handle_(std::make_shared<Handle>(pyclock))
+  py::object pyclock, py::capsule pycontext, int64_t period_nsec)
 {
-  auto clock = clock_handle_->cast<rcl_clock_t *>("rcl_clock_t");
+  clock_handle_ = std::make_shared<rcl_clock_t>(*pyclock.cast<Clock>().rcl_ptr());
```

Comment:
I looks like this is creating a copy of the clock, but I don't think that's desired. If the original copy of the clock goes out of scope, then the clock will be `fini()`'d, and the jump callback pointers on the `rcl_clock_t` itself will point to free'd memory.

Instead, I think `Clock` should have a method that returns an `std::shared_ptr<rcl_clock_t>`, maybe

```c++
std::shared_ptr<rcl_clock_t>
rcl_shared_ptr()
{
    return rcl_clock_
}
```

and then `Timer` can store that with `: clock_handle_(clock.rcl_shared_ptr())`

Reply:
@ahcorde mind addressing this comment?